File: lib/models/post.dart
import 'package:hello_world/models/user.dart';
import 'package:hello_world/models/comment.dart';

class Post {
  String id;
  String title;
  String content;
  String avatar_url;
  List<String> image_urls;
  User user;
  int num_comments;
  String postDate;

  Post(this.id, this.title, this.content, this.avatar_url, this.image_urls, this.user, this.num_comments, this.postDate);
}

File: server/models/post.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;
var mongoosePaginate = require('mongoose-paginate-v2');

let postSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true,
    },
    content: {
        type: String, 
        require: true,
    },
    user: {
        type: Schema.Types.ObjectId, 
            ref: 'User',
            require: true,
    },
    comments:[
        {
            type: Schema.Types.ObjectId, 
            ref: 'Comment',
            require: false,
            default: []
        }
    ],
    date: {
        type: Date,
        require: false,
        default: new Date()
    }
});

postSchema.plugin(mongoosePaginate);
module.exports = mongoose.model('Post', postSchema);